### ISSUE TITLE
[SDK 46] `react-native-maps` outdated, with blocker issue

### ISSUE THREAD

**Issue by BrodaNoel** [expo/expo#18547]
### Summary

The problem here is that SDK 44 was using `react-native-maps@0.29.4`. That version was working pretty good.
Now I'm upgrading to SDK 46 (jumping 2 version in a row), and the new version is `0.31.1`, which includes a big issue that was fixed in version `1.0.0`.

Right now, `react-native-maps` is on `1.2.0`.

You can check the full report of this issue in https://github.com/react-native-maps/react-native-maps/discussions/4378

What's the plan for a situation like this? Should we wait 3 months until SDK 47 is released with `1.2.0`?

The problem here is that the issue is kind of blocker. You can't define the `initialRegion` on a map in some situations. So basically you can't load the map in the correct area of the world. Kind of a blocker.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

46

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 11.6.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.4.0 - ~/.nvm/versions/node/v16.4.0/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 7.18.1 - ~/.nvm/versions/node/v16.4.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 27, 31
        Build Tools: 23.0.1, 23.0.3, 25.0.3, 26.0.1, 27.0.3, 28.0.3, 29.0.2
        System Images: android-31 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^46.0.0 => 46.0.2
      react: 18.0.0 => 18.0.0
      react-native: 0.69.3 => 0.69.3
    npmGlobalPackages:
      eas-cli: 0.57.0
      expo-cli: 6.0.1
    Expo Workflow: managed

### Reproducible demo

Full details in the [mentioned](https://github.com/react-native-maps/react-native-maps/discussions/4378) issue

**Comment by BrodaNoel** [comment]
Guess who had the best weekend...

After 3 days of debug, I cherry picked all the changes from version `0.31.1` to version `1.2.0`, and created a patch, which seems to be working.

**file**: `react-native-maps+0.31.1.patch`

```
diff --git a/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m b/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
index 01227ed..8a4215c 100644
--- a/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
+++ b/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
@@ -62,9 +62,11 @@ id regionAsJSON(MKCoordinateRegion region) {
   NSMutableArray<UIView *> *_reactSubviews;
   MKCoordinateRegion _initialRegion;
   MKCoordinateRegion _region;
-  BOOL _initialCameraSetOnLoad;
+  BOOL _initialRegionSet;
+  BOOL _initialCameraSet;
+  BOOL _didLayoutSubviews;
+  BOOL _didPrepareMap;
   BOOL _didCallOnMapReady;
-  BOOL _didMoveToWindow;
   BOOL _zoomTapEnabled;
 }
 
@@ -83,9 +85,11 @@ id regionAsJSON(MKCoordinateRegion region) {
     _cameraProp = nil;
     _initialRegion = MKCoordinateRegionMake(CLLocationCoordinate2DMake(0.0, 0.0), MKCoordinateSpanMake(0.0, 0.0));
     _region = MKCoordinateRegionMake(CLLocationCoordinate2DMake(0.0, 0.0), MKCoordinateSpanMake(0.0, 0.0));
-    _initialCameraSetOnLoad = false;
+    _initialRegionSet = false;
+    _initialCameraSet = false;
+    _didLayoutSubviews = false;
+    _didPrepareMap = false;
     _didCallOnMapReady = false;
-    _didMoveToWindow = false;
     _zoomTapEnabled = YES;
 
     // Listen to the myLocation property of GMSMapView.
@@ -245,42 +249,47 @@ id regionAsJSON(MKCoordinateRegion region) {
     ];
 }
 
-- (void)didMoveToWindow {
-  if (_didMoveToWindow) return;
-  _didMoveToWindow = true;
+- (void)layoutSubviews {
+  [super layoutSubviews];
+  if(_didLayoutSubviews) return;
+  _didLayoutSubviews = true;
 
   if (_initialCamera != nil) {
     self.camera = _initialCamera;
+    _initialCameraSet = true;
   }
   else if (_initialRegion.span.latitudeDelta != 0.0 &&
       _initialRegion.span.longitudeDelta != 0.0) {
     self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self andMKCoordinateRegion:_initialRegion];
+    _initialRegionSet = true;
   } else if (_region.span.latitudeDelta != 0.0 &&
       _region.span.longitudeDelta != 0.0) {
     self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self andMKCoordinateRegion:_region];
   }
-
-  [super didMoveToWindow];
 }
 
 - (void)setInitialRegion:(MKCoordinateRegion)initialRegion {
-  if (_initialCameraSetOnLoad) return;
   _initialRegion = initialRegion;
-  _initialCameraSetOnLoad = _didMoveToWindow;
-  self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self andMKCoordinateRegion:initialRegion];
+  if(!_initialRegionSet && _didLayoutSubviews){
+    self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self andMKCoordinateRegion:initialRegion];
+    _initialRegionSet = true;
+  }
 }
 
 - (void)setInitialCamera:(GMSCameraPosition*)initialCamera {
-    if (_initialCameraSetOnLoad) return;
     _initialCamera = initialCamera;
-    _initialCameraSetOnLoad = _didMoveToWindow;
-    self.camera = initialCamera;
+    if(!_initialCameraSet && _didLayoutSubviews){
+      self.camera = initialCamera;
+      _initialCameraSet = true;
+    }
 }
 
 - (void)setRegion:(MKCoordinateRegion)region {
   // TODO: The JS component is repeatedly setting region unnecessarily. We might want to deal with that in here.
   _region = region;
-  self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self  andMKCoordinateRegion:region];
+  if(_didLayoutSubviews) {
+    self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self  andMKCoordinateRegion:region];
+  }
 }
 
 - (void)setCameraProp:(GMSCameraPosition*)camera {
@@ -288,14 +297,23 @@ id regionAsJSON(MKCoordinateRegion region) {
     self.camera = camera;
 }
 
+- (void)setOnMapReady:(RCTBubblingEventBlock)onMapReady {
+    _onMapReady = onMapReady;
+    if(!_didCallOnMapReady && _didPrepareMap) {
+      self.onMapReady(@{});
+      _didCallOnMapReady = true;
+    }
+}
 
 - (void)didPrepareMap {
   UIView* mapView = [self valueForKey:@"mapView"]; //GMSVectorMapView
   [self overrideGestureRecognizersForView:mapView];
 
-  if (_didCallOnMapReady) return;
-  _didCallOnMapReady = true;
-  if (self.onMapReady) self.onMapReady(@{});
+  if (!_didCallOnMapReady && self.onMapReady) {
+    self.onMapReady(@{});
+    _didCallOnMapReady = true;
+  }
+  _didPrepareMap = true;
 }
 
 - (void)mapViewDidFinishTileRendering {

```

This could be a valid workaround while we wait for Expo SDK 47 with `react-native-maps@1.2.0`.

If somebody doesn't know how to apply it, please read about `yarn patch-package` (`yarn patch-package react-native-maps`)

**Comment by brentvatne** [comment]
we updated react-native-maps in https://github.com/expo/expo/pull/18052 to the latest version at the time.

you don't need to wait, if you use eas build just install the version that you want and ignore the warning that you see about it not being the recommended version for sdk 46. we likely want to provide a mechanism that lets you opt out of these warnings because this is a perfectly normal thing to do if you want to make the deliberate choice to do so

**Comment by BrodaNoel** [comment]
😟 What a wasted weekend...

**Comment by brentvatne** [comment]
sorry. we update to the latest version of external libraries when they seem to be stable, but that isn't always the case and we can't fully QA all of them in every context - we have to depend on what the developers of those libraries ship.

to follow up on this i created this task internally

<img width="614" alt="image" src="https://user-images.githubusercontent.com/90494/183322990-b5689251-6d6f-4bb8-af5e-ac25c69992f2.png">

**Comment by BrodaNoel** [comment]
@brentvatne but look what I got if I install `1.2.0`

<img width="1162" alt="Screen Shot 2022-08-08 at 00 41 42" src="https://user-images.githubusercontent.com/1837969/183334470-32fe78b2-0f5d-4a2f-a260-503bd7225234.png">
 
<img width="1440" alt="Screen Shot 2022-08-08 at 00 42 27" src="https://user-images.githubusercontent.com/1837969/183334560-b4053785-4567-458a-a860-f04aa4f5d327.png">

**Comment by brentvatne** [comment]
pass in the --clear-cache flag to eas build cmd

**Comment by BrodaNoel** [comment]
@brentvatne nop... `eas build --profile development -p ios --clear-cache`.
Same error.

<img width="1440" alt="Screen Shot 2022-08-08 at 15 00 23" src="https://user-images.githubusercontent.com/1837969/183483344-eebed7fb-3427-4a00-b43f-31e6831e878f.png">

**Comment by BrodaNoel** [comment]
Is build ID enough? `3e3cb36c-88fb-4dcb-bfcc-21a1e13412f4`

**Comment by mvrahas** [comment]
I ran into the same issue with react-native-maps (0.31.1) when upgrading from Expo 44 to Expo 46. 

I tried installing newer versions of react-native-maps and running `eas build --profile development -p ios --clear-cache`. I keep getting the same error as @BrodaNoel 

I noticed that the newer versions of react-native-maps require a minimum deployment target of >= iOS 13.0. My expo project is set at the default 12.4. Is there an easy way to increase this in the managed workflow? I tried setting up a config plugin for it but the documentation is confusing to navigate.

The only solution I can think of is reverting back to Expo 44. Unfortunately, the bugs with react-native-maps (0.31.1) are causing a lot of problems in my app.

Here's the ids for some of my failed builds...
827d11ad-07d2-4d4a-8525-79d5252c8167
12fd2ea2-629b-40a0-a171-f5d8c1da4b12
d5908f84-5de5-4e68-8689-610ec8a78655
